core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -330,8 +330,18 @@ protected CompletionStage<Void> pollStoreAvailability() {
       try {
          AggregateCompletionStage<Void> stageBuilder = CompletionStages.aggregateCompletionStage();
          for (StoreStatus storeStatus : stores) {
-            CompletionStage<Boolean> availableStage = storeStatus.store.isAvailable();
-            stageBuilder.dependsOn(availableStage.thenCompose(isAvailable -> {
+            CompletionStage<Boolean> availableStage;
+            try {
+               availableStage = storeStatus.store.isAvailable();
+            } catch (Throwable t) {
+               log.storeIsAvailableCheckThrewException(t, storeStatus.store.getClass().getName());
+               availableStage = CompletableFutures.booleanStage(false);
+            }
+            stageBuilder.dependsOn(availableStage.handle((isAvailable, throwable) -> {
```

Comment:
I think you need to use `handle + thenCompose` or `exceptionally + thenCompose` (later makes more sense to me), otherwise `stageBuilder` won't wait for the returned `CompletableFuture`.

Reply:
Thanks. I went with the latter :+1: